### PR TITLE
[FrameworkBundle] Fix paths with % in it (like urlencoded)

### LINKED ISSUE #17306: Paths with % in them does not supported
Getting an exception when calling cache:clear on path with '%'. My build server urlencodes branches when deploing them to stage, so paths become like `/var/www/projects/test/feature%2Ftest_branch/`

```

  [Symfony\Component\Debug\Exception\ContextErrorException]
  Warning: sprintf(): Too few arguments



Exception trace:
 () at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:664
 Symfony\Component\Debug\ErrorHandler->handleError() at n/a:n/a
 sprintf() at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:664
 Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension->registerTranslatorConfiguration() at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:112
 Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension->load() at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php:55
 Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationPass->process() at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/MergeExtensionConfigurationPass.php:39
 Symfony\Component\HttpKernel\DependencyInjection\MergeExtensionConfigurationPass->process() at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php:117
 Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:614
 Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /var/www/projects/test/feature%2Ftest_branch/app/bootstrap.php.cache:2565
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /var/www/projects/test/feature%2Ftest_branch/app/bootstrap.php.cache:2344
 Symfony\Component\HttpKernel\Kernel->boot() at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:70
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /var/www/projects/test/feature%2Ftest_branch/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /var/www/projects/test/feature%2Ftest_branch/app/console:27
```

### PR DESCRIPTION
**Pull request by scaytrase**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | Travis passed |
| Fixed tickets | #17306 |
| License | MIT |
| Doc PR | none |

### CODE REVIEW

**Comment by dosten** [comment]
This is a BC break, you need to manually escape the % char

**Comment by scaytrase** [comment]
@dosten  How should this be a BC break? Kernel parameter `%` signs are resolved before. And all other `%` should be passed as is to the sprintf. In other case they cause Exception (see the bug attached)

**Comment by scaytrase** [comment]
i.e if we have path `/var/www/some%2Fproject/` (which is `'/var/www/' . urlencode('some/project')`) we will have the exception. Passign `%` as is should not be a BC break as sprintf expects only one `%s` template in the path.

**Comment by Tobion** [comment]
The better fix is to only apply the `sprintf` to `/Resources/%s/translations` and not the kernel root dir. This way there is no need to encode anything.

**Comment by scaytrase** [comment]
@Tobion Yes, and this would have more clear logic with sprifntf. 

Does this one looks better?

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -562,13 +562,13 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
                 $dirs[] = dirname($r->getFileName()).'/../../Resources/translations';
             }
         }
-        $overridePath = $container->getParameter('kernel.root_dir').'/Resources/%s/translations';
+        $rootDir = $container->getParameter('kernel.root_dir');
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: line 575 can reuse $rootDir

Comment (scaytrase) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: @Tobion updated

**Comment by Tobion** [comment]
:+1: 

Status: Reviewed

**Comment by fabpot** [comment]
Thank you @scaytrase.

**Pull request merged**